Il2CppInspector.Common/IL2CPP/Il2CppInspector.cs

```diff
@@ -294,6 +294,27 @@ private List<MetadataUsage> buildMetadataUsages()
             return Binary.MethodInvokerIndices[module][methodInModule - 1];
         }
 
+        public MetadataUsage[] GetVTable(Il2CppTypeDefinition definition) {
+            MetadataUsage[] res = new MetadataUsage[definition.vtable_count];
+            for (int i = 0; i < definition.vtable_count; i++) {
+                var encodedIndex = VTableMethodIndices[definition.vtableStart + i];
+                uint index;
+                MetadataUsageType usageType;
+                if (Version < 19) {
+                    var flag = encodedIndex & 0x80000000;
+                    index = Binary.VTableMethodReferences[encodedIndex & 0x7FFFFFFF];
+                    usageType = (flag != 0) ? MetadataUsageType.MethodRef : MetadataUsageType.MethodDef;
+                } else {
+                    var encodedType = encodedIndex & 0xE0000000;
+                    usageType = (MetadataUsageType)(encodedType >> 29);
+                    index = encodedIndex & 0x1FFFFFFF;
+                }
```

Comment:
Replace lines 301--311 with a static method in `MetadataUsage` with a signature along the lines of:

````cs
public static (uint index, MetadataUsageType usageType) GetDecodedIndex(uint encodedIndex)
```

Make a call to it from `buildMetadataUsages` using `encodedSourceIndex` to avoid repetition.

Other than that, happy with this PR :)

Reply:
OK, I added a patch that's a little different, but in the same spirit as what you mentioned. I found that `DestinationIndex` is not used anywhere usefully, so I ditched it, then made your proposed method return a whole `MetadataUsage` instead. I think it's OK - feel free to review it.

Reply:
Reviewed, good eye on `DestinationIndex`, I hadn't noticed that. I'm happy, merging :)